File: backend/flight_booking_project/src/main/java/com/javaweb/flight_booking_project/dto/request/FlightRequest.java
package com.javaweb.flight_booking_project.dto.request;


import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.Date;

@Data
public class FlightRequest {
    @NotNull
//    @JsonProperty("departure_airport")
    private Long departureAirport;

    @NotNull
//    @JsonProperty("arrival_airport")
    private Long arrivalAirport;

    @NotNull
//    @JsonProperty("departure_time")
    private String departureTime;

    @NotNull
//    @JsonProperty("seat_class")
    private String seatClass;
}

File: mobile/lib/data/model/flight_request.dart
class FlightRequest {
  int departureAirport;
  int arrivalAirport;
  String departureTime;
  String seatClass;

  FlightRequest({
    required this.departureAirport,
    required this.arrivalAirport,
    required this.departureTime,
    required this.seatClass,
  });

  factory FlightRequest.fromJson(Map<String, dynamic> json) {
    return FlightRequest(
      departureAirport: json['departureAirport'],
      arrivalAirport: json['arrivalAirport'],
      departureTime: json['departureTime'],
      seatClass: json['seatClass'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'departureAirport': departureAirport,
      'arrivalAirport': arrivalAirport,
      'departureTime': departureTime,
      'seatClass': seatClass,
    };
  }
}
